server/ollamacfg/config.go

```diff
@@ -0,0 +1,167 @@
+package ollamacfg
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type Config struct {
+	AllowOrigins []string
+	Debug        bool
+	DemandLLMLib string
+	MaxRunners   int
+	MaxVRAM      uint64
+	NoPrune      bool
+	NumParallel  int
+	RunnersDir   string
+	TmpDir       string
+
+	loaded bool
+}
+
+func (cfg *Config) LogValue() slog.Value {
+	return slog.GroupValue(
+		slog.Any("OLLAMA_ORIGINS", cfg.AllowOrigins),
+		slog.Bool("OLLAMA_DEBUG", cfg.Debug),
+		slog.String("OLLAMA_LLM_LIBRARY", cfg.DemandLLMLib),
+		slog.Int("OLLAMA_MAX_LOADED_MODELS", cfg.MaxRunners),
+		slog.Uint64("OLLAMA_MAX_VRAM", cfg.MaxVRAM),
+		slog.Bool("OLLAMA_NOPRUNE", cfg.NoPrune),
+		slog.Int("OLLAMA_NUM_PARALLEL", cfg.NumParallel),
+		slog.String("OLLAMA_RUNNERS_DIR", cfg.RunnersDir),
+		slog.String("OLLAMA_TMPDIR", cfg.TmpDir),
+	)
+}
+
+var cfg Config
+var cfgMu sync.Mutex
+
+var defaultAllowOrigins = []string{
+	"localhost",
+	"127.0.0.1",
+	"0.0.0.0",
+}
+
+// Clean quotes and spaces from the value
+func getEnv(key string) string {
+	return strings.Trim(os.Getenv(key), "\"' ")
+}
+
+func initCfg() {
```

Comment:
This may be an anti-pattern but could we use `init()`?

Reply:
I wanted a bit more control on when the vars were loaded so we could have unit test cases that do `os.Setenv(...)` then grab init or call other routines that need variables init'd.  If we use `init()` then it would read the var before we updated it.

Reply:
Refactored to do both - primarily `init()` driven with an exposed function to reload if we want to fiddle with env vars at runtime and have them respected.